### DISCUSSION TITLE
Overhead of vitest skewing benchmark results

### DISCUSSION THREAD

**Discussion by kyscott18** [wevm/viem#1093] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.7.0

### Current Behavior

Overhead of vitest seems to be slowing down the runtime such that maximum performance isn't being achieved. I was working on a application level ethereum sdk to replace @uniswap/sdk-core, and the performance gains that I thought would evidently show up, weren't there. A really quick poc is benchmarking noop vs add. In vitest there is often no difference. My intuition is that functions that are small enough get drowned out by the overhead of vitest.

### Expected Behavior

I think switching benchmarking frameworks to mitata, should give much more accurate (and more importantly favorable) results to viem over ethers. 

### Steps To Reproduce

<img width="819" alt="Screenshot 2023-08-26 at 2 51 01 PM" src="https://github.com/wagmi-dev/viem/assets/43524469/d5755761-1683-4f1b-a9b3-04e96edb9c7a">

```js

import { bench, describe } from "vitest";

describe("", () => {
	bench("noop", () => {});

	bench("add", () => {
		1 + 2;
	});
});
```

With mitata:
<img width="615" alt="Screenshot 2023-08-26 at 2 52 21 PM" src="https://github.com/wagmi-dev/viem/assets/43524469/868980ea-8a09-4054-b9dd-ada458895965">

```js
import { bench, group, run } from "mitata";

group("", () => {
	bench("noop", () => {});
	bench("add", () => {
		1 + 2;
	});
});

await run();
```

Notice the much larger iteration times but also the ops/sec when run using mitata is a few orders of magnitudes greater than when using vitest.

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://github.com/kyscott18/vitest-overhead-example

**Comment by jxom** [comment]
Ha. Interesting! Thanks for flagging. Will investigate more this week.

**Comment by kyscott18** [comment]
I can try to migrate to mitata if you think it would be useful. Took me like 15 minutes on a different repo cause the abi is pretty similar to vitest.

**Comment by jxom** [comment]
Do you think this is a config issue? What if we increase the warmup in Vitest?

**Comment by kyscott18** [comment]
I don't think its that because of the huge discrepancy but I will play around with it when I get some more time. It took my benchmarks from 2x to 30x so I thought it might be interesting for viem.

**Comment by tmm** [comment]
Might be worth posting findings here too https://github.com/vitest-dev/vitest/issues/2220

**Comment by Dunqing** [comment]
Look like the `mitata` benchmark result is incorrect. In my local, If I switch the order of `noop` and `add`, the speed is reversed. That's obviously not true.


`noop` precedes `add`.
<img width="634" alt="image" src="https://github.com/wagmi-dev/viem/assets/29533304/982beffe-2469-4725-a72f-4f65ca34aa29">

`add` precedes `noop`.
<img width="597" alt="image" src="https://github.com/wagmi-dev/viem/assets/29533304/de30eff6-1a8f-415c-b48d-1d8dee3a725a">
